apis/v1beta1/shared_types.go

```diff
@@ -21,9 +21,10 @@ import (
 )
 
 // ParentReference identifies an API object (usually a Gateway) that can be considered
-// a parent of this resource (usually a route). The only kind of parent resource
-// with "Core" support is Gateway. This API may be extended in the future to
-// support additional kinds of parent resources, such as HTTPRoute.
+// a parent of this resource (usually a route). The only kinds of parent resources
+// with "Core" support are Service (only for implementations supporting the Mesh
+// conformance profile) and Gateway. This API may be extended in the future to
+// support additional kinds of parent resources.
```

Comment:
Maybe something like this would line up well with our conformance profiles approach? 

/cc @shaneutt @mlavacca 

```suggestion
// a parent of this resource (usually a route). There are two kinds of parent resources
// with "Core" support:
//
// * Gateway (Gateway conformance profile)
// * Service (Mesh conformance profile)
//
// This API may be extended in the future to support additional kinds of parent
// resources.
```

Reply:
Yea, I struggled a bit figuring out how to introduce a docs pattern to express this well, would definitely appreciate more eyes on this to try to get it right.

Reply:
yeah, I think this description fits very well 👍